pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+func newPackagePublishCmd() *cobra.Command {
+
+	var err error
+	var path string
+	var tag string
+
+	cmd := &cobra.Command{
+		Use:   "publish-sdk <package> <language>",
+		Args:  cobra.ExactArgs(2),
+		Short: "Publish a package SDK to supported package registries.",
+		Long:  "Publish a package SDK to supported package registries.",
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Validate the language
+
+			lang := args[1]
+
+			switch lang {
+			case "all":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "nodejs":
+				fmt.Println("Nodejs is supported, publishing")
+			case "python":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "java":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "dotnet":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			default:
+				return fmt.Errorf(
+					"language %s is not supported. Must be one of: \n"+
+						"- all \n"+
+						"- nodejs \n"+
+						"- python \n"+
+						"- java \n"+
+						"- dotnet ", lang,
+				)
+			}
+
+			return nil
+		},
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			pkg := args[0]
+			lang := args[1]
+
+			err = publish(pkg, lang, path, tag)
+			if err != nil {
+				return fmt.Errorf("error publishing: %s", err.Error())
+			}
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	cmd.PersistentFlags().StringVar(&tag, "tag", "",
+		`The tag for your NPM package, such as "latest""`)
+	return cmd
+}
+
+func publish(pkg, lang, path, tag string) error {
+
+	fmt.Printf("You are publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+	case "python":
+		publishToPyPI(pkg)
+	case "java":
+		publishToMaven(pkg)
+	case "dotnet":
+		publishToNuget(pkg)
+	default:
+		// Publish to all registries
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+		publishToPyPI(pkg)
+		publishToMaven(pkg)
+		publishToNuget(pkg)
```

Comment:
We don't know at this point whether all publish functions will be having the same signature. This sounds like a refactor once we know what each language needs.

Reply:
Makes sense! This was speculative suggestion hoping to get something else:
I want to avoid repetition of publish calls between language-specific cases and "all".
I don't want to risk having a language handled in the specific case, but not in "all", or if a language ends up with more complex logic, I want us to avoid having to duplicate that.

If the signatures don't yet match, we can still get this through a different means.
What do you think about something like the following?

```go
func publish(..., lang string) error {
  enabled := func(l string) bool {
    return lang == l || lang == "all"
  }

  if enabled("nodejs") {
    if err := publishNPM(...); err != nil {
      // ...
    }
  }
  if enabled("python") {
    publishPython(..)
  }
```

It's not the cleanest, but it'll meet this goal while leaving flexibility in function signature.